<?xml version="1.0"?>


<launch>  
    
<!--<node name="master" pkg="armadillo2_bgu" type="master_old" output="screen">
    </node> -->
   
<node name="master" pkg="armadillo2_bgu" type="master" output="screen">
    </node>

<!--<node name="omer_planner" pkg="armadillo2_bgu" type="omer_planner" output="screen">
   </node>  --> 
    
    <!-- <node name="run_node" pkg="armadillo2_bgu" type="run_object_rec.py" output="log">
    </node> --> 

    <arg name="move" default="false"/>
    <arg name="moveit" default="false"/>
    <arg name="moveit_gazebo" default="false"/>
    <arg name="manipulate" default="false"/>
    <arg name="vision" default="false"/>
    
    <group if="$(arg move)" >
        <node name="move" pkg="armadillo2_bgu" type="move" output="screen"></node>
	<node name="moveNear" pkg="armadillo2_bgu" type="moveNear" output="screen"></node>
    </group>

    <group if="$(arg moveit_gazebo)" >
        <node name="pick_gazebo" pkg="armadillo2_bgu" type="pick_gazebo" output="screen"></node>
        <node name="arm_pick" pkg="armadillo2_bgu" type="arm_pick" output="screen"></node>
    </group>

    <group if="$(arg moveit)" >
        <node name="pick" pkg="armadillo2_bgu" type="pick" output="screen"></node>
       <node name="arm_pick" pkg="armadillo2_bgu" type="arm_pick" output="screen"></node>
    </group>

    <group if="$(arg manipulate)" >
    <node name="manipulation" pkg="armadillo2_bgu" type="manipulation" output="screen"></node>
    <node name="give" pkg="armadillo2_bgu" type="give" output="screen"></node>
    </group>
     
    <group if="$(arg vision)" >
        <node name="find_objects_node" pkg="object_detection" type="find_objects_node" output="screen"></node>
        <node name="search_for_person" pkg="armadillo2_bgu" type="search_for_person" output="screen"></node>
    </group>

 


    
</launch>
